get-sysinternals-du.js

```diff
@@ -5,45 +5,82 @@ const path = require('path')
 const decompress = require('decompress')
 const { HttpsProxyAgent } = require('https-proxy-agent')
 
-// Only run for Windows
-if (process.platform !== 'win32') {
-  process.exit(0)
-}
+exports.workspace = __dirname
 
-// check if du is already installed
-if (
-  fs.existsSync(
-    path.join(__dirname, 'bin', `du${process.arch === 'x64' ? '64' : ''}.exe`)
-  )
-) {
-  process.exit(0)
+exports.onDuZipDownloaded = function (tempFilePath) {
+  decompress(tempFilePath, path.join(exports.workspace, 'bin'))
 }
 
-const duZipLocation =
-  process.env.FAST_FOLDER_SIZE_DU_ZIP_LOCATION ||
-  'https://download.sysinternals.com/files/DU.zip'
+exports.downloadDuZip = function (mirror) {
+  const duZipLocation =
+    mirror || 'https://download.sysinternals.com/files/DU.zip'
 
-// checks for proxy variables in user environment
-const proxyAddress =
-  process.env.HTTPS_PROXY ||
-  process.env.https_proxy ||
-  process.env.HTTP_PROXY ||
-  process.env.http_proxy
+  // checks for proxy variables in user environment
+  const proxyAddress =
+    process.env.HTTPS_PROXY ||
+    process.env.https_proxy ||
+    process.env.HTTP_PROXY ||
+    process.env.http_proxy
 
-if (proxyAddress) {
-  const agent = new HttpsProxyAgent(proxyAddress)
+  if (proxyAddress) {
+    https.globalAgent = new HttpsProxyAgent(proxyAddress)
+  }
 
-  https.globalAgent = agent
-}
+  console.log(`downloading du.zip from ${duZipLocation}`)
+  if (!mirror) {
+    console.log(
+      `if you have trouble while downloading, try set process.env.FAST_FOLDER_SIZE_DU_ZIP_LOCATION to a proper mirror or local file path`
+    )
+  }
 
-https.get(duZipLocation, function (res) {
-  const tempFilePath = path.join(os.tmpdir(), 'du.zip')
+  https.get(duZipLocation, function (res) {
+    const tempFilePath = path.join(os.tmpdir(), 'du.zip')
 
-  const fileStream = fs.createWriteStream(tempFilePath)
-  res.pipe(fileStream)
+    const fileStream = fs.createWriteStream(tempFilePath)
+    res.pipe(fileStream)
 
-  fileStream.on('finish', function () {
-    fileStream.close()
-    decompress(tempFilePath, path.join(__dirname, 'bin'))
+    fileStream.on('finish', function () {
+      fileStream.close()
+      exports.onDuZipDownloaded(tempFilePath)
+    })
   })
-})
+}
+
+exports.default = function () {
+  // Only run for Windows
+  if (process.platform !== 'win32') {
+    return
+  }
+
+  // check if du is already installed
+  let duBinFilename = `du${process.arch === 'x64' ? '64' : ''}.exe`
+  let defaultDuBinPath = path.join(exports.workspace, 'bin', duBinFilename)
```

Comment:
please use const for variables that won't change

Reply:
This should be done by eslint, should we add the "prefer-const" rule ?

Reply:
Done.